gluoncv/data/transforms/block.py

```diff
@@ -62,18 +62,13 @@ def __init__(self, size, pad=None, interpolation=2):
         if isinstance(size, numeric_types):
             size = (size, size)
         self._args = (size, interpolation)
-        if isinstance(pad, int):
-            self.pad = ((pad, pad), (pad, pad), (0, 0))
-        else:
-            self.pad = pad
-
-    def forward(self, x):
-        if self.pad:
-            x_pad = np.pad(x.asnumpy(), self.pad,
-                           mode='constant', constant_values=0)
-
-        return image.random_crop(nd.array(x_pad), *self._args)[0]
-
+        self.pad = ((pad, pad), (pad, pad), (0, 0)) if isinstance(pad, int) else pad
+        self.forward = self.forward_pad if self.pad else self.forward_nopad
```

Comment:
can you avoid using monkey patch (self.forward_pad self.forward_nopad) but switch inside self.forward? it might be easier to catch error for single `forward` function

Reply:
fixed.
I just think using a monkey patch is better than using if since it should be faster.